### PR TITLE
Fixed currency/number not working in filters of trigger handlers

### LINKED ISSUE #539: Filter Field on Trigger Handler doesn't work with Number/Currency fields
If you select a Number, Currency, or any field that uses numeric data type (i.e. percentage, rollup summary, formula) the TDTM filtering doesn't work. The class is still run.

**Steps to Reproduce**

1. Create custom field on Contact such as Household Income (Currency 18,0)
2. Set the Filter Field on a Contact-related Trigger Handler (such as CON_DoNotContact_TDTM) to Household_Income__c
3. Set the Filter Value to 50000
4. Create or update a Contact with 50000 in the Household Income field
5. Check Deceased and Save
6. Notice that the trigger runs even though there is a filter

I tested the above with a simple Number field and a Rollup Summary field that returns a number and got the same result.

### PR DESCRIPTION
**Pull request by xinnanlisf**
# Critical Changes

# Changes
- Added support for currency, integer, double, and percent fields in the trigger handler Filter Field.

# Issues Closed
Fixes #539
# New Metadata

# Deleted Metadata

# Testing Notes
1. Create custom field on Contact such as Household Income (Currency 18,0)
2. Set the Filter Field on a Contact-related Trigger Handler (such as CON_DoNotContact_TDTM) to Household_Income__c
3. Set the Filter Value to 50000
4. Create or update a Contact with 50000 in the Household Income field
5. Check Deceased and Save
6. Notice that the trigger runs even though there is a filter

### CODE REVIEW

**Comment by xinnanlisf** [comment]
@lianeyee 

Could you take a look at the release notes please? Thank you! :)

**Comment by lianeyee** [comment]
@xinnanlisf  Like I often say, I think we should mention more specifics. =) "Added support for currency, integer, double, and percent fields in trigger handler filters."

**Comment by xinnanlisf** [comment]
@lianeyee 

Ah my bad, sorry :)

**Comment by lianeyee** [comment]
No worries. Sounds like that works so let's go with that. I'll ask @sam-knox to take a look, too.

File: src/classes/TDTM_Filter.cls [hunk 1]
```diff
@@ -414,12 +414,20 @@ public class TDTM_Filter {
...
         } else if(fieldType == DisplayType.Reference) {
             return ID.valueOf(val);
+        } else if(fieldType == DisplayType.CURRENCY) {
+            return Decimal.valueOf(val);
+        } else if(fieldType == DisplayType.INTEGER) {
+            return Integer.valueOf(val);
+        } else if(fieldType == DisplayType.DOUBLE) {
+            return Double.valueOf(val);
+        } else if(fieldType == DisplayType.PERCENT) {
+            return Decimal.valueOf(val);
```
Comment (weizsforg) [inline src/classes/TDTM_Filter.cls]: Ideally we need the unit test to test the Numeric field.  Can we add some test cases for one/more new supported type? Maybe we can use some Account Standard field for filtering.

**Comment by weizsforg** [comment]
@xinnanlisf Looks good. 
Ideally we need the unit test to test the Numeric field. Can we add some test cases for one/more new supported type? Maybe we can use some Account Standard field for filtering.

**Comment by xinnanlisf** [comment]
@weizsforg 

Good catch. I will go ahead and work on unit tests.

**Comment by weizsforg** [comment]
@xinnanlisf Thanks for adding the unit test.

**Comment by annejackson** [comment]
@xinnanlisf could you give me a number (field) example for the steps to test to make sure I check that one correctly as well? I can also test that making sure triggers work still even if they don't have filter feeds filled out (which I'm guessing for the trigger mentioned above I would just have to check deceased to get this trigger to run)? Anything else you can think of to test?

**Comment by annejackson** [comment]
@xinnanlisf , @ceiroa  and @weizsforg  looks like some checks haven't been completed for this PR yet, could you all check?

**Comment by weizsforg** [comment]
@annejackson Looks like the Test is still queued for this PR maybe due to the recent merge. It might need some time to complete the test.

**Comment by annejackson** [comment]
I believe I can also do this test step to check whether the filter runs or not (since the acct_ind account trigger is easy to verify), please verify and check @xinnanlisf 
1. Create custom field on Account such as Household Income (Currency 18,0)
(make sure default account model is administrative in Heda Settings)
2. Set the Filter Field on an Account-related Trigger Handler (such as ACCT_Individual Accounts_TDTM) to Household_Income__c
3. Set the Filter Value to 50000
4. Create (and later on test update) a Contact with 50000 in the Household Income field and Save
5. Notice that the trigger runs even though there is a filter. 
With this fix, the account name shouldn't get updated on create or save. (trigger shouldn't run).

**Comment by annejackson** [comment]
@xinnanlisf @ceiroa 
Error:
1. Created contact with household income value of 50000 so the do not contact trigger would  notrun.
2. Edited that contact - last name field and also changed the household income to 60000. saved
Got this message. see screenshot.
Review the errors on this page.
TDTM_Contact: execution of BeforeUpdate caused by: System.ListException: List index out of bounds: 0 Class.CON_DoNotContact_TDTM.run: line 62, column 1 Class.TDTM_TriggerHandler.runClass: line 143, column 1 Class.TDTM_TriggerHandler.run: line 75, column 1 Class.TDTM_Global_API.run: line 61, column 1 Trigger.TDTM_Contact: line 33, column 1

error on update

**Comment by annejackson** [comment]
@xinnanlisf  reminded me that the above error I found was the issue fixed in the other PR:
https://github.com/SalesforceFoundation/HEDAP/issues/538, fix 538 just wasn't in my scratch org- so I'll retest that scenario on the sandbox for the release. Testing has passed and currency/number (percentage) is now working in filters of trigger handlers. Merging this pull request.

**Pull request merged**